cocotb/share/makefiles/Makefile.pylib.Msys

```diff
@@ -35,3 +35,13 @@ ifeq ($(PYTHON_LIBDIR),None)
 endif
 
 PYTHON_INCLUDEDIR := $(shell $(PYTHON_BIN) -c 'import distutils.sysconfig; print( distutils.sysconfig.get_python_inc() )')
+
+# On Windows there is no support for @rpath/LD_LIBRARY_PATH so we need to set PATH to find cocotb libraries
+# https://docs.microsoft.com/en-us/windows/win32/dlls/dynamic-link-library-search-order?redirectedfrom=MSDN#standard-search-order-for-desktop-applications
+export PATH := $(LIB_DIR):$(PATH)
+
+# Set PYTHONHOME to properly populate sys.path in embedded python interpreter
+export PYTHONHOME := $(shell $(PYTHON_BIN) -c 'from distutils.sysconfig import get_config_var; print(get_config_var("prefix"))')
```

Comment:
Shouldn't we do this on all platforms?

Also, the docs seem to suggest that PYTHONHOME is set automatically based on the python interpreter used. Can you confirm it fails without it?

Reply:
>Shouldn't we do this on all platforms?

I think would not harm and possibly help.

>Can you confirm it fails without it?

On Windows yes. It will fail to `import site` as the first error.

Reply:
From [the python docs](https://docs.python.org/3.9/using/windows.html#finding-modules):

> For those who want to bundle Python into their application or distribution, the following advice will prevent conflicts with other installations:
> 
> * Include a ._pth file alongside your executable containing the directories to include. This will ignore paths listed in the registry and environment variables, and also ignore site unless import site is listed.
> 
> * If you are loading python3.dll or python37.dll in your own executable, explicitly call Py_SetPath() or (at least) Py_SetProgramName() before Py_Initialize().
> 
> * Clear and/or overwrite PYTHONPATH and set PYTHONHOME before launching python.exe from your application.
> 
> * If you cannot use the previous suggestions (for example, you are a distribution that allows people to run python.exe directly), ensure that the landmark file (Lib\os.py) exists in your install directory. (Note that it will not be detected inside a ZIP file, but a correctly named ZIP file will be detected instead.)

So I think this is ok for now, although calling `Py_SetProgramName` may make more sense in future.

Reply:
> I think would not harm and possibly help.

I think we should do this then.

Reply:
You did this, then force-pushed a change that removed it.

Reply:
Should be ok?